### PR TITLE
OBJ Model Loader now loads textures properly, and supports per-vertex coloring.

### PR DESCRIPTION
**Pull request by shadekiller666**

### CODE REVIEW

**Comment by LordIllyohs** [comment]
Why make a new PR for this?

**Comment by shadekiller666** [comment]
Because i did things out of order... :/

**Comment by shadekiller666** [comment]
Other PR has been closed, this one was made automatically when i pushed.

**Comment by Abastro** [comment]
So is this the only active PR now? Then squash your commits.

File: src/test/java/net/minecraftforge/debug/ModelLoaderRegistryDebug.java [hunk 1]
```diff
@@ -1,23 +1,18 @@
 package net.minecraftforge.debug;
 
 import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
 import java.util.List;
 
+import javax.vecmath.Vector3f;
+
+import scala.actors.threadpool.Arrays;
```
Comment (loordgek) [inline src/test/java/net/minecraftforge/debug/ModelLoaderRegistryDebug.java]: scala?

**Comment by shadekiller666** [comment]
scala is not supposed to be there, let me fix that

**Comment by cpw** [comment]
Can you make sure you've uptaken the latest fixes from the 1.7.10 branch? I think there's some float fixes in there.. Thanks!

**Comment by ZeroNoRyouki** [comment]
uhm I may be wrong but there is something I don't get in the OBJModel:retexture method. You are creating new Texture instances for the changed textures but you are not storing those and then you use the model current, unchanged, material library to create the new model

should not you make a copy of the current material library, modify it and then use it to create the new OBJModel?

**Comment by shadekiller666** [comment]
Zero, you are correct, i forgot to use that new Texture to replace the one in the material library, thanks for pointing that out.

**Comment by ZeroNoRyouki** [comment]
I’m glad I could be of help :)

I've just checked your revised code and I think that there is still a problem regarding the material library as you are modifing the original model matlib instead of creating a modified copy for the new model

Let's say I have an OBJ with texture A, B and C and I create a first OBJModel from it. Then I create 3 more models from this first model changing the textures, something like this (skipping the ImmutableMap and just using strings for brevity):


IModel default = <get a reference to the default model>;
IModel modelA = default.retexture("A", "A1"); // get a new model using texture A1 instead of A
IModel modelB = default.retexture("B", "B1"); // get a new model using texture B1 instead of B
IModel modelC = default.retexture("C", "C1"); // get a new model using texture C1 instead of C


modelA will get textures A1/B/C but modelB will get A1/B1/C instead of A/B1/C and modelC will end up with A1/B1/C1 instead of A/B/C1

**Comment by shadekiller666** [comment]
This is not a problem any longer, a copy of matLib is made within the retexture() function.

**Comment by ZeroNoRyouki** [comment]
Sorry to bother you again, but did you make a new PR ? The code in this one was not changed, it still modifiy/use the current model matLib to create the new model

**Comment by shadekiller666** [comment]
Haven't pushed, adding another example block and things aren't cooperating.

**Comment by tambry** [comment]
Any news if this is getting or when it will be merged?

**Comment by shadekiller666** [comment]
It will be pulled, all I can say is soon as for when, there are a few things I need to work out, but it is almost finished.

**Comment by tambry** [comment]
@shadekiller666 Thank you for the status update.

**Pull request closed without merging**